chemprop/uncertainty/evaluator.py

```diff
@@ -67,9 +148,28 @@ def evaluate(self, preds: Tensor, uncs: Tensor, targets: Tensor, mask: Tensor) -
 
 @UncertaintyEvaluatorRegistry.register("spearman")
 class SpearmanEvaluator(UncertaintyEvaluator):
+    """
+    Evaluate the Spearman rank correlation coefficient between the uncertainties and errors in the model predictions.
+
+    The correlation coefficient returns a value in the [-1, 1] range, with better scores closer to 1
+    observed when the uncertainty values are predictive of the rank ordering of the errors in the model prediction.
+    """
+
     def evaluate(self, preds: Tensor, uncs: Tensor, targets: Tensor, mask: Tensor) -> Tensor:
-        ...
-        return
+        spearman_coeffs = []
+        for j in range(uncs.shape[1]):
+            mask_j = mask[:, j]
+            preds_j = preds[:, j]
+            targets_j = targets[:, j]
+            uncs_j = uncs[:, j]
+            masked_preds = preds_j[mask_j]
+            masked_targets = targets_j[mask_j]
+            masked_uncs = uncs_j[mask_j]
+            masked_errs = (masked_preds - masked_targets).abs()
+            spearman = SpearmanCorrCoef()
+            spearman_coeff = spearman(masked_uncs, masked_errs)
+            spearman_coeffs.append(spearman_coeff)
```

Comment:
Please modify it if you take the above suggestions

Reply:
I have modified it to be consistent with the NLL evaluators.